MINOR: remove zk from several tests

cmccabe opened the pull request:
Remove ZK from
 - core/src/test/scala/integration/kafka/admin/RemoteTopicCrudTest.scala
 - core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala
 - core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala
 - tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java

cmccabe committed:
MINOR: remove zk from several tests

Remove ZK from
 - core/src/test/scala/integration/kafka/admin/RemoteTopicCrudTest.scala
 - core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala
 - core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala
 - tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java
 - core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala

chia7712 commented on core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala:
```diff
@@ -191,8 +182,9 @@ class AddPartitionsTest extends BaseRequestTest {
+  @Disabled
   @ParameterizedTest
-  @ValueSource(strings = Array("zk")) // TODO: add kraft support
+  @ValueSource(strings = Array("kraft")) // TODO: add kraft support
```
We need to revisit this test to make it stable under KRaft. I've opened Jira https://issues.apache.org/jira/browse/KAFKA-18098 to address it. Could you please revert this change?

peterxcli replied:
Hi @chia7712, thanks for addressing this.
PR is ready! https://github.com/apache/kafka/pull/17955

cmccabe replied:
sure, let's address this in #17955. I'll remove it from this PR. Thanks @peterxcli and @chia7712

chia7712 commented on core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala:
```diff
@@ -217,8 +209,9 @@ class AddPartitionsTest extends BaseRequestTest {
+  @Disabled
   @ParameterizedTest
-  @ValueSource(strings = Array("zk")) // TODO: add kraft support
+  @ValueSource(strings = Array("kraft")) // TODO: add kraft support
```
ditto

cmccabe replied:
sure, let's address this in #17955. I'll remove it from this PR. Thanks @peterxcli and @chia7712

chia7712 reviewed (commented):
@cmccabe thanks for this patch. Could you please fix the conflicts?

cmccabe committed:
Merge branch 'trunk' into cmccabe_remove_zk_in_several_tests

cmccabe commented:
- Removed modifications to `core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala` and removed it from the PR description
- Added reference to `core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala` to the PR description.

chia7712 commented on tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java:
```diff
@@ -106,13 +115,13 @@ public void setUp(TestInfo testInfo) {
         startSasl(jaasSections(KAFKA_SERVER_SASL_MECHANISMS, Some$.MODULE$.apply(KAFKA_CLIENT_SASL_MECHANISM), Both$.MODULE$,
             JaasTestUtils.KAFKA_SERVER_CONTEXT_NAME));
         super.setUp(testInfo);
```
This fails with latest code ... I have filed https://issues.apache.org/jira/browse/KAFKA-18109 to fix it.
